api/nodes/serializers.py

```diff
@@ -46,7 +63,14 @@ def get_absolute_url(self, obj):
         return obj.absolute_url
 
     def get_node_count(self, obj):
-        return len(obj.nodes)
+        request = self.context['request']
+        user = request.user
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        nodes = [node for node in obj.nodes if node.can_view(auth) and node.primary]
```

Comment:
This seems like a responsibility of one of the Filter classes defined above, not something the serializer should be doing.